src/screens/TopicPageScreen/index.js

```diff
@@ -162,10 +162,10 @@ const TopicPageScreen = (props) => {
     try {
       const resultTopicDetail = await getTopics(domain);
       if (resultTopicDetail.data) {
-        const detail = resultTopicDetail.data[0];
+        const detail = resultTopicDetail.data.find((item) => item.name === domain);
         setTopicDetail(detail);
-        handleFollowData(detail.is_followed_by);
-        setMemberCount(Number(detail.followersCount));
+        handleFollowData(detail?.is_followed_by);
+        setMemberCount(Number(detail?.followersCount));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a more robust way of finding the `detail` object within the `resultTopicDetail.data` array. It also uses optional chaining when accessing properties of `detail`, which is a good practice to prevent potential undefined or null errors. However, there's no error handling if `detail` is not found in the array. This could lead to issues if `detail` is expected to exist but doesn't. Consider adding error handling for this case.
```diff
+        const detail = resultTopicDetail.data.find((item) => item.name === domain);
+        if (!detail) {
+          throw new Error(`Detail not found for domain: ${domain}`);
+        }
         setTopicDetail(detail);
         handleFollowData(detail?.is_followed_by);
         setMemberCount(Number(detail?.followersCount));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->